zio-http/src/main/scala/zio/http/Flash.scala

```diff
@@ -0,0 +1,390 @@
+   * to `Html`.
+   *
+   * Usage e.g.: `Flash.getMessageHtml.foldHtml(showNotice, showAlert)(_ ++ _)`
+   */
+  def getMessageHtml: Flash[Message[Html, Html]] =
+    getMessageHtml[String, String](a => Dom.text(a), b => Dom.text(b))
+
+  /**
+   * Gets any flash value of type `A` with the given key `key`.
+   */
+  def get[A: Schema](key: String): Flash[A] = Flash.Get(Schema[A], key)
+
+  /**
+   * Gets a flash value of type `String` with the given key `key`.
+   */
+  def getString(key: String): Flash[String] = get[String](key)
+
+  /**
+   * Gets a flash value of type `A` associated with the notice key.
+   */
+  def getNotice[A: Schema]: Flash[A] = get[A](Message.Notice.name)
+
+  /**
+   * Gets a flash value of type `A` associated with the alert key.
+   */
+  def getAlert[A: Schema]: Flash[A] = get[A](Message.Alert.name)
+
+  /**
+   * Gets a flash value of type `Float` with the given key `key`.
+   */
+  def getFloat(key: String): Flash[Float] = get[Float](key)
+
+  /**
+   * Gets a flash value of type `Double` with the given key `key`.
+   */
+  def getDouble(key: String): Flash[Double] = get[Double](key)
+
+  /**
+   * Gets a flash value of type `Int` with the given key `key`.
+   */
+  def getInt(key: String): Flash[Int] = get[Int](key)
+
+  /**
+   * Gets a flash value of type `Long` with the given key `key`.
+   */
+  def getLong(key: String): Flash[Long] = get[Long](key)
+
+  /**
+   * Gets a flash value of type `UUID` with the given key `key`.
+   */
+  def getUUID(key: String): Flash[UUID] = get[UUID](key)
+
+  /**
+   * Gets a flash value of type `Boolean` with the given key `key`.
+   */
+  def getBoolean(key: String): Flash[Boolean] = get[Boolean](key)
+
+  /**
+   * Gets the first flash value of type `A` regardless of any key.
+   */
+  def get[A: Schema]: Flash[A] = withInput { map =>
+    map.keys.map(a => Flash.get(a)(Schema[A])).reduce(_ <> _)
+  }
+
+  private def loop[A](flash: Flash[A], map: Map[String, String]): Either[Throwable, A] =
+    flash match {
+      case Get(schema, key)   =>
+        map.get(key).toRight(new Throwable(s"no flash key: $key")).flatMap { value =>
+          JsonCodec.jsonDecoder(schema).decodeJson(value).left.map(e => new Throwable(e))
+        }
+      case WithInput(f)       =>
+        loop(f(map), map)
+      case OrElse(self, that) =>
+        (loop(self, map) match {
+          case Left(_)      => loop(that, map)
+          case Right(value) => Right(value)
+        }).asInstanceOf[Either[Throwable, A]]
+      case FlatMap(self, f)   =>
+        loop(self, map) match {
+          case Right(value) => loop(f(value), map)
+          case Left(e)      => Left(e)
+        }
+      case Succeed(a)         => Right(a)
+      case Fail(message)      => Left(new Throwable(message))
+    }
+
+  private[http] def run[A](flash: Flash[A], sourceRequest: Request): Either[Throwable, A] = {
+    sourceRequest
+      .cookie(COOKIE_NAME)
+      .toRight(new Throwable("flash cookie doesn't exist"))
```

Comment:
🟢 I always prefer to use the Java exceptions are they're supposed to be used. `RuntimeException` is supposed to be used in this case, not `Throwable`, as it's a runtime exception
```suggestion
      .toRight(new RuntimeException("flash cookie doesn't exist"))
```

Reply:
@TomTriple You should probably do the same with all your `Throwable` usages